File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/api/user_card.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import PlusApiResultBoolean, PlusApiResultListPlusUserCardVO, PlusApiResultPagePlusUserCardVO, PlusApiResultPlusUserCardVO, PlusUserCardForm, QueryListForm

class UserCardApi:
    """user_card API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def update(self, body: PlusUserCardForm) -> PlusApiResultPlusUserCardVO:
        """Update user-card binding"""
        return self._client.put(f"/backend/v3/api/user/card", json=body)

    def create(self, body: PlusUserCardForm) -> PlusApiResultPlusUserCardVO:
        """Create user-card binding"""
        return self._client.post(f"/backend/v3/api/user/card", json=body)

    def list_by_page(self, body: Optional[QueryListForm] = None, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPagePlusUserCardVO:
        """Get user-card bindings by page"""
        return self._client.post(f"/backend/v3/api/user/card/list", json=body, params=params)

    def list_all_entities(self, body: Optional[QueryListForm] = None) -> PlusApiResultListPlusUserCardVO:
        """Get all user-card bindings"""
        return self._client.post(f"/backend/v3/api/user/card/list/all", json=body)

    def get_by_id(self, id: str) -> PlusApiResultPlusUserCardVO:
        """Get user-card binding by ID"""
        return self._client.get(f"/backend/v3/api/user/card/{id}")

    def delete(self, id: str) -> PlusApiResultBoolean:
        """Delete user-card binding"""
        return self._client.delete(f"/backend/v3/api/user/card/{id}")

File: sdkwork-backend-sdk-flutter/lib/src/api/user_card.dart
import '../http/client.dart';
import '../models.dart';

class UserCardApi {
  final HttpClient _client;
  
  UserCardApi(this._client);

  /// Update user-card binding
  Future<PlusApiResultPlusUserCardVO?> update(PlusUserCardForm body) async {
    final response = await _client.put(ApiPaths.backendPath('/user/card'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusUserCardVO ? response : null;
  }

  /// Create user-card binding
  Future<PlusApiResultPlusUserCardVO?> create(PlusUserCardForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/user/card'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusUserCardVO ? response : null;
  }

  /// Get user-card bindings by page
  Future<PlusApiResultPagePlusUserCardVO?> listByPage(QueryListForm? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.backendPath('/user/card/list'), body: body, params: params, contentType: 'application/json');
    return response is PlusApiResultPagePlusUserCardVO ? response : null;
  }

  /// Get all user-card bindings
  Future<PlusApiResultListPlusUserCardVO?> listAllEntities(QueryListForm? body) async {
    final response = await _client.post(ApiPaths.backendPath('/user/card/list/all'), body: body, contentType: 'application/json');
    return response is PlusApiResultListPlusUserCardVO ? response : null;
  }

  /// Get user-card binding by ID
  Future<PlusApiResultPlusUserCardVO?> getById(String id) async {
    final response = await _client.get(ApiPaths.backendPath('/user/card/${id}'));
    return response is PlusApiResultPlusUserCardVO ? response : null;
  }

  /// Delete user-card binding
  Future<PlusApiResultBoolean?> delete(String id) async {
    final response = await _client.delete(ApiPaths.backendPath('/user/card/${id}'));
    return response is PlusApiResultBoolean ? response : null;
  }
}
